packages/TimePicker/src/TimePicker.js

```diff
@@ -11,120 +11,169 @@
+  defaultValue: PropTypes.string,
+
+  /** If the TimePicker is disabled. */
+  isDisabled: PropTypes.bool,
+
+  /** The label shown for the AM. */
+  labelAM: PropTypes.string,
+
+  /** The label shown for the Custom. */
+  labelCustom: PropTypes.string,
+
+  /** The label shown for the hours. */
+  labelHours: PropTypes.string,
+
+  /** The label shown for the minutes. */
+  labelMinutes: PropTypes.string,
 
-  constructor(props) {
-    super(props);
-    this.name = `${props.prefix}${btoa(Math.random()).substring(0, 8)}`;
-    this.state = {
+  /** The label shown for the period. */
+  labelPeriod: PropTypes.string,
+
+  /** The label shown for the PM. */
+  labelPM: PropTypes.string,
+
+  /** Callback to be executed when the value is changed. */
+  onChange: PropTypes.func,
+
+  /** Callback to be executed when there is an error. */
+  onError: PropTypes.func,
+
+  /** */
+  prefix: PropTypes.string,
+};
+
+const defaultProps = {
+  a11yText: "Time (hh:mm)",
+  defaultIsOpen: false,
+  defaultValue: null,
+  labelCustom: "Custom",
+  labelHours: "Hours",
+  labelMinutes: "Minutes",
+  labelPeriod: "Period",
+  labelAM: "am",
+  labelPM: "pm",
+  isDisabled: false,
+  onChange: () => {},
+  onError: () => {},
+  prefix: "timePicker",
+};
+
+function TimePicker(props) {
+  const {
+    a11yText,
+    defaultIsOpen,
+    defaultValue,
+    labelCustom,
+    labelHours,
+    labelMinutes,
+    labelPeriod,
+    labelAM,
+    labelPM,
+    isDisabled,
+    onChange,
+    onError,
+    prefix,
+    ...moreProps
+  } = props;
+
+  const [isOpen, setIsOpen] = React.useState(defaultIsOpen);
+  const [time, setTime] = React.useState(() => {
+    if (defaultValue) {
+      const defaultTime = getExplodeTime(defaultValue);
+      return {
+        hh: defaultTime.hh,
+        mm: defaultTime.mm,
+        period: defaultTime.period,
+      };
+    }
+    return {
       hh: null,
-      isPristine: true,
-      isVisible: this.props.defaultIsVisible,
       mm: null,
       period: null,
-      timeStr: null,
-      isTabIndexActive: 0,
-      value: "",
     };
-  }
+  });
+  const [timeStr, setTimeStr] = React.useState(null);
+  const [isTabIndexActive, setIsTabIndexActive] = React.useState(0);
+  const [value, setValue] = React.useState(defaultValue);
 
-  static getDerivedStateFromProps(props, state) {
-    if (props.defaultValue && !state.value && state.isPristine) {
-      const time = getExplodeTime(props.defaultValue);
-
-      if (time.error) {
-        return {
-          ...time,
-          error: props.labelError,
-          isVisible: false,
-          isPristine: false,
-        };
+  React.useEffect(() => {
+    if (defaultValue) {
+      const { hh, mm, period, error, timeStr: newTimeStr, value: newValue } = getExplodeTime(defaultValue);
+
+      if (error) {
+        setIsOpen(false);
+        // TODO: not required if we assume consumer always pass in valid value
+        // onError(error);
```

Comment:
Hi @nahumzs could you take a look at this part? Can we assume consumers always pass in a valid `defaultValue`? Cause I don't want to include `onError` as the dependency of this `useEffect`

Reply:
I'm wondering if might exists the case where we allowed the user to saved a wrong date? 🤔, anyway, if that is the case then we can update the component to support that scenario. 

But what can actually happen is when you are testing/playing with the component you might pass an incorrect date I guess with a simple  `console.error()` or `console.warn` informing the consumer the date is incorrect might be enough for now.

Reply:
Oh sounds good, I like the idea of using `console.error`